### PR TITLE
Security update

### PR DESCRIPTION
**Pull request by awholegunch**
Running as a Quick Build: http://builds.cask.co/browse/CDAP-DQB300-2

Pages of interest:
- Admin Manual > Security > Impersonation http://builds.cask.co/artifact/CDAP-DQB300/shared/build-2/Docs-HTML/4.1.0/en/admin-manual/security/impersonation.html#admin-impersonation
- Dev Manual > Building Blocks > Datasets http://builds.cask.co/artifact/CDAP-DQB300/shared/build-2/Docs-HTML/4.1.0/en/developers-manual/building-blocks/datasets/index.html
- Dev Manual > Security http://builds.cask.co/artifact/CDAP-DQB300/shared/build-2/Docs-HTML/4.1.0/en/developers-manual/security/index.html

### CODE REVIEW

File: cdap-docs/admin-manual/source/security/impersonation.rst [hunk 1]
```diff
@@ -48,6 +61,113 @@ described in the installation section for each distribution (:ref:`Cloudera Mana
...
+To configure Hive to be able to impersonate other users, set in ``hive-site.xml`` the property::
+
+  <property>
+      <name>hive.server2.enable.doAs</name>
+      <value>true</value>
+  </property>
+
+Note that the CDAP Explore service ignores this setting and needs to be able to
+impersonate users who can create and access entities in CDAP. This can by done by adding
+properties in your ``core-site.xml``. The first property allows CDAP to impersonate users
```
Comment (Nishith) [inline cdap-docs/admin-manual/source/security/impersonation.rst]: The first property allows hive to impersonate users belonging to the listed groups.

File: cdap-docs/developers-manual/source/security/index.rst [hunk 2]
```diff
@@ -73,3 +73,24 @@ If the standard authentication mechanisms are not sufficient, you can provide a
...
+.. rubric:: Impersonation
+
+Impersonation allows users to run programs and access datasets, streams, and other
+resources as pre-configured users (a *principal*). Currently, CDAP supports configuring
+impersonation at a namespace and at an application level, with application level
+configuration having a higher precedence than namespace level.
+
+If impersonation is enabled, and you don't specify a principal for an application, stream,
+or dataset, then the namespace owner's principal is used. If there is no namespace owner
+or you are using the default namespace, then the default principal is used (as set by the
```
Comment (Nishith) [inline cdap-docs/developers-manual/source/security/index.rst:89]: Please verify that we are using "default principal" to refer to these two settings at other places too.

File: cdap-docs/reference-manual/source/http-restful-api/dataset.rst [hunk 3]
```diff
@@ -123,10 +124,11 @@ with JSON-formatted name of the dataset type, properties, and description in a b
      - ``PUT /v3/namespaces/default/data/datasets/mydataset``
    * - Body
      - ``{"typeName":"co.cask.cdap.api.dataset.table.Table",`` ``"properties":{"dataset.table.ttl":"3600"},``
-       ``"description":"My Dataset Description",`` ``"principal":"user/somehost.net@somekdc.net"}``
+       ``"description":"My Dataset Description",`` ``"principal":"user/example.net@examplekdc.net"}``
```
Comment (Nishith) [inline cdap-docs/reference-manual/source/http-restful-api/dataset.rst]: By convention, the realm, "examplekdc.net" in this example, is all caps.

File: cdap-docs/reference-manual/source/http-restful-api/dataset.rst [hunk 4]
```diff
@@ -123,10 +124,11 @@ with JSON-formatted name of the dataset type, properties, and description in a b
...
    * - Description
      - Creates a dataset named *mydataset* of the type ``Table`` in the namespace *default*
-       with the time-to-live property set to 1 hour and a description of ``My Dataset Description``
+       with the time-to-live property set to 1 hour, a description of ``My Dataset Description``,
+       owned by the principal identified by ``user/example.net@examplekdc.net``.
```
Comment (Nishith) [inline cdap-docs/reference-manual/source/http-restful-api/dataset.rst]: Realm in all caps.

File: cdap-docs/reference-manual/source/http-restful-api/dataset.rst [hunk 5]
```diff
@@ -237,7 +239,7 @@ The response body will contain JSON-formatted metadata of the existing dataset::
          }
        ]
      },
-     "principal": "user/somehost.net@somekdc.net"
+     "principal": "user/example.net@examplekdc.net"
```
Comment (Nishith) [inline cdap-docs/reference-manual/source/http-restful-api/dataset.rst]: Same here.

File: cdap-docs/reference-manual/source/http-restful-api/dataset.rst (hunk 3 above)
Comment (awholegunch) [reply cdap-docs/reference-manual/source/http-restful-api/dataset.rst]: Fixed.

File: cdap-docs/reference-manual/source/http-restful-api/dataset.rst (hunk 4 above)
Comment (awholegunch) [reply cdap-docs/reference-manual/source/http-restful-api/dataset.rst]: Fixed.

File: cdap-docs/reference-manual/source/http-restful-api/dataset.rst (hunk 5 above)
Comment (awholegunch) [reply cdap-docs/reference-manual/source/http-restful-api/dataset.rst]: Fixed.

File: cdap-docs/admin-manual/source/security/impersonation.rst (hunk 1 above)
Comment (awholegunch) [reply cdap-docs/admin-manual/source/security/impersonation.rst]: Fixed.

File: cdap-docs/developers-manual/source/security/index.rst (hunk 2 above)
Comment (awholegunch) [reply cdap-docs/developers-manual/source/security/index.rst:89]: This text is inserted elsewhere, so the other places where it is used are exactly the same:

- Building Blocks > Streams
- Building Blocks > Datasets

**Review by Nishith — APPROVED** [review]

**Pull request merged**